build/ninja_gen/src/bin/update_uv.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright: Ankitects Pty Ltd and contributors
+// License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
+
+use std::error::Error;
+use std::fs;
+use std::path::Path;
+
+use regex::Regex;
+use reqwest::blocking::Client;
+use serde_json::Value;
+
+fn fetch_uv_release_info() -> Result<String, Box<dyn Error>> {
+    let client = Client::new();
+
+    println!("Fetching latest uv release info from GitHub...");
+    // Fetch latest release info
+    let response = client
+        .get("https://api.github.com/repos/astral-sh/uv/releases/latest")
+        .header("User-Agent", "Anki-Build-Script")
+        .send()?;
```

Comment:
Probably not an issue since I don't see this being use programmatically anywhere, IIUC, but worth noting:

> The primary rate limit for unauthenticated requests is 60 requests per hour.

https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#primary-rate-limit-for-unauthenticated-users

Reply:
We're only likely to run it once every few months, so I think we'll be fine :-)